test/api/application/services/test_authorisation_service.py

```diff
@@ -117,341 +157,661 @@ def test_raises_error_when_browser_makes_request_and_no_user_token_provided(self
+        ],
+    )
+    def test_denies_permissions(
+        self,
+        domain: str,
+        dataset: str,
+        token_permissions: List[str],
+        endpoint_scopes: List[str],
+    ):
+        with pytest.raises(
+            AuthorisationError, match="Not enough permissions to access endpoint"
+        ):
             match_user_permissions(token_permissions, endpoint_scopes, domain, dataset)
 
-    @pytest.mark.parametrize("domain, dataset, token_permissions, endpoint_scopes", [
-        (None, None, ["READ/domain/dataset"], ["READ"]),
-        (None, None, ["WRITE/domain/dataset"], ["WRITE"])])
-    def test_matches_permissions_when_domain_and_dataset_are_none(self,
-                                                                  domain: str,
-                                                                  dataset: str,
-                                                                  token_permissions: List[str],
-                                                                  endpoint_scopes: List[str]):
+    @pytest.mark.parametrize(
+        "domain, dataset, token_permissions, endpoint_scopes",
+        [
+            (None, None, ["READ/domain/dataset"], ["READ"]),
+            (None, None, ["WRITE/domain/dataset"], ["WRITE"]),
+        ],
+    )
+    def test_matches_permissions_when_domain_and_dataset_are_none(
+        self,
+        domain: str,
+        dataset: str,
+        token_permissions: List[str],
+        endpoint_scopes: List[str],
+    ):
         try:
             match_user_permissions(token_permissions, endpoint_scopes, domain, dataset)
         except BaseAppException:
             pytest.fail("Unexpected Unauthorised Error was thrown")
 
-    @pytest.mark.parametrize("domain, dataset, token_permissions, endpoint_scopes", [
-        (None, "dataset", ["READ/domain/dataset"], ["READ"]),
-        ("domain", None, ["WRITE/domain/dataset"], ["WRITE"])])
-    def test_raises_error_when_domain_or_dataset_is_none(self,
-                                                         domain: str,
-                                                         dataset: str,
-                                                         token_permissions: List[str],
-                                                         endpoint_scopes: List[str]):
-        with pytest.raises(AuthorisationError, match="Not enough permissions to access endpoint"):
+    @pytest.mark.parametrize(
+        "domain, dataset, token_permissions, endpoint_scopes",
+        [
+            (None, "dataset", ["READ/domain/dataset"], ["READ"]),
+            ("domain", None, ["WRITE/domain/dataset"], ["WRITE"]),
+        ],
+    )
+    def test_raises_error_when_domain_or_dataset_is_none(
+        self,
+        domain: str,
+        dataset: str,
+        token_permissions: List[str],
+        endpoint_scopes: List[str],
+    ):
+        with pytest.raises(
+            AuthorisationError, match="Not enough permissions to access endpoint"
+        ):
```

Comment:
Noting that this was in the list of files that you said you changed but from a (admittedly quick) look through these are black changes rather than code?

Reply:
Yeah just black changes